### PR TITLE
Added responsive contact form with custom styling

### PR DESCRIPTION
**Pull request by Naveed-SNR**

### CODE REVIEW

**Comment by devdaim6** [comment]
@Naveed-SNR cxe goy preview fail 



gode.karu su theek

**Comment by Naveed-SNR** [comment]
> @Naveed-SNR cxe goy preview fail
> 
> gode.karu su theek

jo issues ma log h whi problem h na sirf

**Comment by Naveed-SNR** [comment]
@devdaim6 check

**Comment by devdaim6** [comment]
image not required and button wech no border

**Comment by Naveed-SNR** [comment]
> 
> 
> image not required and button wech no border

yi chu lakut msle next commit ma thk karu ga

**Comment by Naveed-SNR** [comment]
> > 
> > image not required and button wech no border
> 
> yi chu lakut msle next commit ma thk karu ga

filhal kar  cxe merge

**Comment by devdaim6** [comment]
and i was thinking of adding a select button in contact too , for categorizing website bug realted contact , or information regarding clubs or something like this may b

**Comment by devdaim6** [comment]
Message me ? remove that🤣🤣

**Comment by Naveed-SNR** [comment]
> and i was thinking of adding a select button in contact too , for categorizing website bug realted contact , or information regarding clubs or something like this may b

Ath ha chu teli UI ti shearun. Thav wo message ma hii user likhe ga

**Comment by devdaim6** [comment]
@Naveed-SNR

**Comment by devdaim6** [comment]
box ko nechay kaha amrica beja hai ? su haav ending if box beyi yus contact info chuna su lagav box nebre kani bone kani ....form aur footer ke beech mai

**Comment by Naveed-SNR** [comment]
> 
> 
> box ko nechay kaha amrica beja hai ? su haav ending if box beyi yus contact info chuna su lagav box nebre kani bone kani ....form aur footer ke beech mai

Ye badi screens p ajeeb hota h agr bahar nikale g box k

Ye acha h aise hii

**Comment by devdaim6** [comment]
on submit toast sonner i told you to use and clear form fields to ""

**Comment by Naveed-SNR** [comment]
> on submit toast sonner i told you to use and clear form fields to ""

Sorui kati kar aki doh pga wechav waen su

File: app/components/ContactForm.jsx [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+  const handleFormSubmit = (event) => {
+    event.preventDefault();
+    const formErrors = validateForm();
+
+    if (Object.keys(formErrors).length > 0) {
+      setErrors(formErrors);
+    } else {
+      // Send the form data to your server or API here
+      console.log('Form submitted!');
+      const formData = new FormData();
```
Comment (devdaim6) [inline app/components/ContactForm.jsx:40]: no need to make formData , simply pass the states which we have like this :
```js
body:JSON.stringify({
name,
email,
message,
image:image?image:""
})
```

File: app/components/ContactForm.jsx [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    else if (!nameRegex.test(name)) newErrors.name = 'Invalid name. Only alphabets and spaces are allowed. Minimum length is 3 characters and maximum is 20 characters.';
+
+    if (!email) newErrors.email = 'Email is required';
+    else if (!emailRegex.test(email)) newErrors.email = 'Invalid email address.';
+
+    if (!message) newErrors.message = 'Message is required';
+    else if (message.length > 100) newErrors.message = 'Message should not exceed 100 characters.';
+
+    if (!image) newErrors.image = 'Image is required';
+    else if (!['image/jpeg', 'image/png'].includes(image.type)) newErrors.image = 'Only .jpg and .png files are allowed';
```
Comment (devdaim6) [inline app/components/ContactForm.jsx:26]: Image field is not requried remove the validation for that

File: app/components/ContactForm.jsx [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+                  <label
+                    htmlFor="image"
+                    className="bg-orange-50 cursor-pointer border border-black text-black py-2 px-2 md:px-4 rounded-sm font-medium hover:bg-black hover:text-white transition-all inline-block"
+                  >
+                    Choose File
+                  </label>
+                  <span className="ml-3">{image ? image.name : "No file selected"}</span>
+                  {errors.image && <span className="text-red-500 text-xs block mt-1">{errors.image}</span>}
+                </div>
+                <div className="mb-4">
```
Comment (devdaim6) [inline app/components/ContactForm.jsx:141]: No border for submit button

**Review by devdaim6 — CHANGES_REQUESTED** [review]
Highlighted issues

**Comment by devdaim6** [comment]
image


@TAWSEEFSHAMIM @Naveed-SNR why all the stuff on  a single page ? it should work either conditionally or just new route for the next step but for new routes you have to keeep track of previous state as well

**Comment by Naveed-SNR** [comment]
> image
> 
> 
> @TAWSEEFSHAMIM @Naveed-SNR why all the stuff on  a single page ? it should work either conditionally or just new route for the next step but for new routes you have to keeep track of previous state as well

@TAWSEEFSHAMIM was only in charge of designing the sections. Arranging these is my part of the job and I'll start working on it today.

**Comment by Naveed-SNR** [comment]
@devdaim6 merge this PR, anything that's not working will be fixed later.

**Review by Naveed-SNR — COMMENTED** [review]
Issues with the register page section will be fixed later.

**Comment by devdaim6** [comment]
> Issues with the register page section will be fixed later.

fix that then will merge no big deal......no fun of merging many PRs , just complete each step and merge.......and if you have to implement other  features better would be create a branch for that push that , and will merge that let it be until its complete , coz preview we have how and what it looks like https://department-of-it-git-dev-devdaims-projects.vercel.app/

**Comment by devdaim6** [comment]
just have contact and registration page here from now until we merge this until create new seperate branches for each feature , will be easy to manage / merge and probably no conflicts to resolve

**Comment by Naveed-SNR** [comment]
> > Issues with the register page section will be fixed later.
> 
> fix that then will merge no big deal......no fun of merging many PRs , just complete each step and merge.......and if you have to implement other  features better would be create a branch for that push that , and will merge that let it be until its complete , coz preview we have how and what it looks like https://department-of-it-git-dev-devdaims-projects.vercel.app/

You should merge. Other contributors also have to pull and they may face issues like large no. of conflicts if there are too many commits in a single PR.

**Comment by devdaim6** [comment]
> > > Issues with the register page section will be fixed later.
> > 
> > 
> > fix that then will merge no big deal......no fun of merging many PRs , just complete each step and merge.......and if you have to implement other  features better would be create a branch for that push that , and will merge that let it be until its complete , coz preview we have how and what it looks like https://department-of-it-git-dev-devdaims-projects.vercel.app/
> 
> You should merge. Other contributors also have to pull and they may face issues like large no. of conflicts if there are too many commits in a single PR.

BARADAR THATS WHAT I SAID THEY WILL PULL FROM DEV BRANCH , AND FROM NOW ON CREATE SEPERATE BRANCH FOR EACH THING like events page will have seperate branch clubs etc, just complete this and then will merge it , it wont cause issue dont worry

**Comment by Naveed-SNR** [comment]
> > > > Issues with the register page section will be fixed later.
> > > 
> > > 
> > > fix that then will merge no big deal......no fun of merging many PRs , just complete each step and merge.......and if you have to implement other  features better would be create a branch for that push that , and will merge that let it be until its complete , coz preview we have how and what it looks like https://department-of-it-git-dev-devdaims-projects.vercel.app/
> > 
> > 
> > You should merge. Other contributors also have to pull and they may face issues like large no. of conflicts if there are too many commits in a single PR.
> 
> BARADAR THATS WHAT I SAID THEY WILL PULL FROM DEV BRANCH , AND FROM NOW ON CREATE SEPERATE BRANCH FOR EACH THING like events page will have seperate branch clubs etc, just complete this and then will merge it , it wont cause issue dont worry

We're mostly working on one feature at a time, so the dev branch is enough, no need to create a separate branch for every thing. And convey to others that they have to pull from dev branch, if that's what we're doing.

**Comment by devdaim6** [comment]
> > > > > Issues with the register page section will be fixed later.
> > > > 
> > > > 
> > > > fix that then will merge no big deal......no fun of merging many PRs , just complete each step and merge.......and if you have to implement other  features better would be create a branch for that push that , and will merge that let it be until its complete , coz preview we have how and what it looks like https://department-of-it-git-dev-devdaims-projects.vercel.app/
> > > 
> > > 
> > > You should merge. Other contributors also have to pull and they may face issues like large no. of conflicts if there are too many commits in a single PR.
> > 
> > 
> > BARADAR THATS WHAT I SAID THEY WILL PULL FROM DEV BRANCH , AND FROM NOW ON CREATE SEPERATE BRANCH FOR EACH THING like events page will have seperate branch clubs etc, just complete this and then will merge it , it wont cause issue dont worry
> 
> We're mostly working on one feature at a time, so the dev branch is enough, no need to create a separate branch for every thing. And convey to others that they have to pull from dev branch, if that's what we're doing.


converying doesnt make any diff sometimes we just push new commit and i cant review every single detail in every commit thats why if everyone create seperate branch for seperate stuff that wont cause any issues otherwise i have no issues on working only on dev , but its getting complicated day by day because codebase is filling up , cant review every minute step of every commit change

**Comment by Naveed-SNR** [comment]
> > > > > > Issues with the register page section will be fixed later.
> > > > > 
> > > > > 
> > > > > fix that then will merge no big deal......no fun of merging many PRs , just complete each step and merge.......and if you have to implement other  features better would be create a branch for that push that , and will merge that let it be until its complete , coz preview we have how and what it looks like https://department-of-it-git-dev-devdaims-projects.vercel.app/
> > > > 
> > > > 
> > > > You should merge. Other contributors also have to pull and they may face issues like large no. of conflicts if there are too many commits in a single PR.
> > > 
> > > 
> > > BARADAR THATS WHAT I SAID THEY WILL PULL FROM DEV BRANCH , AND FROM NOW ON CREATE SEPERATE BRANCH FOR EACH THING like events page will have seperate branch clubs etc, just complete this and then will merge it , it wont cause issue dont worry
> > 
> > 
> > We're mostly working on one feature at a time, so the dev branch is enough, no need to create a separate branch for every thing. And convey to others that they have to pull from dev branch, if that's what we're doing.
> 
> converying doesnt make any diff sometimes we just push new commit and i cant review every single detail in every commit thats why if everyone create seperate branch for seperate stuff that wont cause any issues otherwise i have no issues on working only on dev , but its getting complicated day by day because codebase is filling up , cant review every minute step of every commit change

You still have to inform everyone that they've to pull from dev branch and not main (because most of them don't know dev is ahead of main). As far as different branches go, only I am working on something currently so I'll push to dev. If I or someone else starts working on something else, we'll create a different branch.

**Comment by devdaim6** [comment]
> > > > > > > Issues with the register page section will be fixed later.
> > > > > > 
> > > > > > 
> > > > > > fix that then will merge no big deal......no fun of merging many PRs , just complete each step and merge.......and if you have to implement other  features better would be create a branch for that push that , and will merge that let it be until its complete , coz preview we have how and what it looks like https://department-of-it-git-dev-devdaims-projects.vercel.app/
> > > > > 
> > > > > 
> > > > > You should merge. Other contributors also have to pull and they may face issues like large no. of conflicts if there are too many commits in a single PR.
> > > > 
> > > > 
> > > > BARADAR THATS WHAT I SAID THEY WILL PULL FROM DEV BRANCH , AND FROM NOW ON CREATE SEPERATE BRANCH FOR EACH THING like events page will have seperate branch clubs etc, just complete this and then will merge it , it wont cause issue dont worry
> > > 
> > > 
> > > We're mostly working on one feature at a time, so the dev branch is enough, no need to create a separate branch for every thing. And convey to others that they have to pull from dev branch, if that's what we're doing.
> > 
> > 
> > converying doesnt make any diff sometimes we just push new commit and i cant review every single detail in every commit thats why if everyone create seperate branch for seperate stuff that wont cause any issues otherwise i have no issues on working only on dev , but its getting complicated day by day because codebase is filling up , cant review every minute step of every commit change
> 
> You still have to inform everyone that they've to pull from dev branch and not main (because most of them don't know dev is ahead of main). As far as different branches go, only I am working on something currently so I'll push to dev. If I or someone else starts working on something else, we'll create a different branch.

you have tawseef working as well , he did his part on registration part give , now he will create a basic login form : @TAWSEEFSHAMIM  create a branch login , route /auth/login  -> fields = username or email then password bas thats all and have a forgotpassword? link  below that login button bas

**Comment by devdaim6** [comment]
> > > > > > > Issues with the register page section will be fixed later.
> > > > > > 
> > > > > > 
> > > > > > fix that then will merge no big deal......no fun of merging many PRs , just complete each step and merge.......and if you have to implement other  features better would be create a branch for that push that , and will merge that let it be until its complete , coz preview we have how and what it looks like https://department-of-it-git-dev-devdaims-projects.vercel.app/
> > > > > 
> > > > > 
> > > > > You should merge. Other contributors also have to pull and they may face issues like large no. of conflicts if there are too many commits in a single PR.
> > > > 
> > > > 
> > > > BARADAR THATS WHAT I SAID THEY WILL PULL FROM DEV BRANCH , AND FROM NOW ON CREATE SEPERATE BRANCH FOR EACH THING like events page will have seperate branch clubs etc, just complete this and then will merge it , it wont cause issue dont worry
> > > 
> > > 
> > > We're mostly working on one feature at a time, so the dev branch is enough, no need to create a separate branch for every thing. And convey to others that they have to pull from dev branch, if that's what we're doing.
> > 
> > 
> > converying doesnt make any diff sometimes we just push new commit and i cant review every single detail in every commit thats why if everyone create seperate branch for seperate stuff that wont cause any issues otherwise i have no issues on working only on dev , but its getting complicated day by day because codebase is filling up , cant review every minute step of every commit change
> 
> You still have to inform everyone that they've to pull from dev branch and not main (because most of them don't know dev is ahead of main). As far as different branches go, only I am working on something currently so I'll push to dev. If I or someone else starts working on something else, we'll create a different branch.

@Naveed-SNR cxe maklaav contact poore fix kaerith beyi kar registration khatam baki wechov patte

**Comment by devdaim6** [comment]
image


NAVEED first thing , dont craete FormData , sending state sedhay as json string in body and second thing when form is submitted without data , errors are displayed but when user changes name email and message state respective errors should get removed on state change

File: app/components/ContactForm.jsx (hunk 1 above)
Comment (devdaim6) [reply app/components/ContactForm.jsx:39]: > no need to make formData , simply pass the states which we have like this :
> 
> ```js
> body:JSON.stringify({
> name,
> email,
> message,
> image:image?image:""
> })
> ```

YE WECH

**Comment by devdaim6** [comment]
[17:19:00.917] Running build in Washington, D.C., USA (East) – iad1
[17:19:01.029] Cloning github.com/devdaim6/department-of-it (Branch: dev, Commit: 15f5744)
[17:19:01.350] Cloning completed: 320.99ms
[17:19:03.709] Restored build cache
[17:19:03.781] Running "vercel build"
[17:19:04.250] Vercel CLI 35.1.0
[17:19:04.600] Running "install" command: `npm install --legacy-peer-deps`...
[17:19:06.004] 
[17:19:06.004] up to date, audited 452 packages in 1s
[17:19:06.004] 
[17:19:06.004] 144 packages are looking for funding
[17:19:06.004]   run `npm fund` for details
[17:19:06.005] 
[17:19:06.005] found 0 vulnerabilities
[17:19:06.016] Detected Next.js version: 14.2.5
[17:19:06.019] Detected `package-lock.json` generated by npm 7+
[17:19:06.020] Running "npm run build"
[17:19:06.207] 
[17:19:06.207] > department-of-it@0.1.0 build
[17:19:06.207] > next build
[17:19:06.207] 
[17:19:06.882]   ▲ Next.js 14.2.5
[17:19:06.883] 
[17:19:06.960]    Creating an optimized production build ...
[17:19:20.448]  ✓ Compiled successfully
[17:19:20.449]    Linting and checking validity of types ...
[17:19:22.076] 
[17:19:22.076] Failed to compile.
[17:19:22.076] 
[17:19:22.076] ./app/register/clubs.jsx
[17:19:22.076] 71:15  Error: Missing "key" prop for element in iterator  react/jsx-key
[17:19:22.076] 
[17:19:22.077] info  - Need to disable some ESLint rules? Learn more here: https://nextjs.org/docs/basic-features/eslint#disabling-rules
[17:19:22.097] Error: Command "npm run build" exited with 1
[17:19:23.352] 


@Naveed-SNR

**Comment by Naveed-SNR** [comment]
> [17:19:00.917] Running build in Washington, D.C., USA (East) – iad1 [17:19:01.029] Cloning github.com/devdaim6/department-of-it (Branch: dev, Commit: [15f5744](https://github.com/devdaim6/department-of-it/commit/15f5744c00a9924c394c3c2dac22295d426f4585)) [17:19:01.350] Cloning completed: 320.99ms [17:19:03.709] Restored build cache [17:19:03.781] Running "vercel build" [17:19:04.250] Vercel CLI 35.1.0 [17:19:04.600] Running "install" command: `npm install --legacy-peer-deps`... [17:19:06.004] [17:19:06.004] up to date, audited 452 packages in 1s [17:19:06.004] [17:19:06.004] 144 packages are looking for funding [17:19:06.004] run `npm fund` for details [17:19:06.005] [17:19:06.005] found 0 vulnerabilities [17:19:06.016] Detected Next.js version: 14.2.5 [17:19:06.019] Detected `package-lock.json` generated by npm 7+ [17:19:06.020] Running "npm run build" [17:19:06.207] [17:19:06.207] > department-of-it@0.1.0 build [17:19:06.207] > next build [17:19:06.207] [17:19:06.882] ▲ Next.js 14.2.5 [17:19:06.883] [17:19:06.960] Creating an optimized production build ... [17:19:20.448] ✓ Compiled successfully [17:19:20.449] Linting and checking validity of types ... [17:19:22.076] [17:19:22.076] Failed to compile. [17:19:22.076] [17:19:22.076] ./app/register/clubs.jsx [17:19:22.076] 71:15 Error: Missing "key" prop for element in iterator react/jsx-key [17:19:22.076] [17:19:22.077] info - Need to disable some ESLint rules? Learn more here: https://nextjs.org/docs/basic-features/eslint#disabling-rules [17:19:22.097] Error: Command "npm run build" exited with 1 [17:19:23.352]
> 
> @Naveed-SNR



Wait yehi thk kr rha hu

**Review by devdaim6 — APPROVED** [review]

**Comment by devdaim6** [comment]
image


@Naveed-SNR chuyna cxaer aasan gomut : fullname is not updating ath chuyni onchange chalaney

**Comment by devdaim6** [comment]
ath chune typey gasan

**Comment by Naveed-SNR** [comment]
> ath chune typey gasan
> 

Wni hy chalan os

**Comment by Naveed-SNR** [comment]
> ath chune typey gasan
> 

Cxe kyazi 4 lati same msg kermicx

**Comment by devdaim6** [comment]
image


> > ath chune typey gasan
> 
> Cxe kyazi 4 lati same msg kermicx

**Comment by Naveed-SNR** [comment]
> image
> 
> 
> > > ath chune typey gasan
> > 
> > Cxe kyazi 4 lati same msg kermicx
> 
> 

GitHub pareshan isko sync issues h

**Comment by devdaim6** [comment]
wani kar ye issue theek te naev pr kar pat kaerzi kaem events ya clubs kaerzi start banavun mai chu tami hissab tath bakend banavun

**Comment by devdaim6** [comment]
cluubs rozan static su thavav hardcoded

**Comment by devdaim6** [comment]
@Naveed-SNR

**Pull request merged**